core/controllers/reader_test.py

```diff
@@ -1956,7 +1956,45 @@ def test_submit_answer_for_exploration_raises_error_with_no_version(self):
                     exp_domain.EXPLICIT_CLASSIFICATION),
             }, expected_status_int=400
         )
-        self.assertEqual(response['error'], 'NONE EXP VERSION: Answer Submit')
+        self.assertEqual(
+            response['error'], 'Missing key in handler args: version.'
+        )
+
+    def test_submit_answer_for_exp_raises_error_with_no_answer_matching_type(
+        self
+    ):
+        # Load demo exploration.
+        exp_id = '6'
+        exp_services.delete_demo(exp_id)
+        exp_services.load_demo(exp_id)
+        version = 1
+
+        self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
+        self.login(self.VIEWER_EMAIL)
+
+        exploration_dict = self.get_json(
+            '%s/%s' % (feconf.EXPLORATION_INIT_URL_PREFIX, exp_id))
+        state_name_1 = exploration_dict['exploration']['init_state_name']
+
+        response = self.post_json(
+            '/explorehandler/answer_submitted_event/%s' % exp_id,
+            {
+                'old_state_name': state_name_1,
+                'answer': [],
+                'version': version,
+                'client_time_spent_in_secs': 0,
+                'session_id': '1PZTCw9JY8y-8lqBeuoJS2ILZMxa5m8N',
+                'answer_group_index': 0,
+                'rule_spec_index': 0,
+                'classification_categorization': (
+                    exp_domain.EXPLICIT_CLASSIFICATION),
+            }, expected_status_int=400
+        )
+        self.assertEqual(
+            response['error'],
+            '''Schema validation for \'answer\' failed: 
+            Type of [] is not present in options'''
+        )
```

Comment:
@Nik-09 how do I split this line into two lines with a line break?

Reply:
Approach 1: Store the error message in a different variable, and then use it here.
Approach 2: Use single quotes (instead of triple) and then split it like -- 
```
'some arbitrary text'
'next line'
```